spaceship/lib/spaceship/portal/certificate.rb

```diff
@@ -312,16 +335,27 @@ def create!(csr: nil, bundle_id: nil)
 
           # look up the app_id by the bundle_id
           if bundle_id
-            app = portal_type.set_client(client).find(bundle_id)
-            raise "Could not find app with bundle id '#{bundle_id}'" unless app
-            app_id = app.app_id
+            if bundle_id.start_with?('merchant.') # ApplePay
+              merchant = Spaceship::Portal.merchant.find(bundle_id)
+              raise "Could not find merchant with bundle id '#{bundle_id}" unless merchant
+              merchant_id = merchant.merchant_id
+            else
+              app = portal_type.set_client(client).find(bundle_id)
+              raise "Could not find app with bundle id '#{bundle_id}'" unless app
+              app_id = app.app_id
+            end
           end
 
           # ensure csr is a OpenSSL::X509::Request
           csr = OpenSSL::X509::Request.new(csr) if csr.kind_of?(String)
 
           # if this succeeds, we need to save the .cer and the private key in keychain access or wherever they go in linux
-          response = client.create_certificate!(type, csr.to_pem, app_id, mac)
+          if type == '4APLUP237T' # = ApplePay
```

Comment:
Maybe instead of adding a comment, we could re-use previously declared key:

```
if type == IOS_CERTIFICATE_TYPE_IDS.key('ApplePayCertificate')
```

⚠️ ☝️ Most probably requires some tweaking, didn't test it out in the code.

Reply:
If that works, that would definitely be nice. Can you test it and report back?

Reply:
@RishabhTayal don't you mind testing the suggestion out? 🙏